### PR TITLE
[stable/mariadb] add metrics servicemonitor

### PR DESCRIPTION
**Pull request by zh168654**
#### What this PR does / why we need it:

Add support for prometheus operator ServiceMonitor.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

File: stable/mariadb/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: mariadb
-version: 6.2.1
+version: 6.2.2
```
Comment (juan131) [inline stable/mariadb/Chart.yaml]: Increase minor version:
```suggestion
version: 6.3.0
```

File: stable/mariadb/templates/servicemonitors.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+{{ if and .Values.metrics.enabled .Values.metrics.serviceMonitor.enabled }}
```
Comment (juan131) [inline stable/mariadb/templates/servicemonitors.yaml]:
```suggestion
{{- if and .Values.metrics.enabled .Values.metrics.serviceMonitor.enabled }}
```

File: stable/mariadb/templates/servicemonitors.yaml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    release: {{ .Release.Name | quote }}
+    heritage: {{ .Release.Service | quote }}  
+    {{- range $key, $value := .Values.metrics.serviceMonitor.selector }}
+    {{ $key }}: {{ $value | quote }}
+    {{- end }}
+spec:
+  selector:
+    matchLabels:
+      app: {{ template "mariadb.name" . }}
+      release: {{ .Release.Name }}
```
Comment (juan131) [inline stable/mariadb/templates/servicemonitors.yaml]: Be consistent with the way you declared the labels before:
```suggestion
      release: {{ .Release.Name | quote }}
```

File: stable/mariadb/templates/servicemonitors.yaml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    {{- if .Values.metrics.serviceMonitor.interval }}
+    interval: {{ .Values.metrics.serviceMonitor.interval }}
+    {{- end }}
+    {{- if .Values.metrics.serviceMonitor.scrapeTimeout }}
+    scrapeTimeout: {{ .Values.metrics.serviceMonitor.scrapeTimeout }}
+    {{- end }}
+  namespaceSelector:
+    matchNames:
+    - {{ .Release.Namespace }}
+{{ end }}
```
Comment (juan131) [inline stable/mariadb/templates/servicemonitors.yaml]:
```suggestion
{{- end }}
```

**Review by juan131 — CHANGES_REQUESTED** [review]
Thanks so much for your contribution, please take a look to my suggestions.

- Document all the new parameters in the **README.md**
- I'd rename the file from `servicemonitors.yaml` to `servicemonitor.yaml`

**Comment by zh168654** [comment]
> Thanks so much for your contribution, please take a look to my suggestions.
> 
> * Document all the new parameters in the **README.md**
> * I'd rename the file from `servicemonitors.yaml` to `servicemonitor.yaml`

@juan131 Thanks for your feedback and suggestions. 
I have followed your recommendation and modified the commits

File: stable/mariadb/README.md [hunk 5]
```diff
@@ -147,6 +147,10 @@ The following table lists the configurable parameters of the MariaDB chart and t
 | `metrics.image.tag`                       | Exporter image tag                                  | `v0.10.0`                                                         |
 | `metrics.image.pullPolicy`                | Exporter image pull policy                          | `IfNotPresent`                                                    |
 | `metrics.resources`                       | Exporter resource requests/limit                    | `nil`                                                             |
+| `metrics.serviceMonitor.enabled`          | if `true`, creates a Prometheus Operator ServiceMonitor (also requires `metrics.enabled` to be `true`)                    | `nil`                                                             |
```
Comment (juan131) [inline stable/mariadb/README.md]: You set the default value to `nil` but it is actually `false`

**Review by juan131 — CHANGES_REQUESTED** [review]
A minor change

**Comment by zh168654** [comment]
@juan131 Thanks, and sorry for the mistake and some previous wrong behaviour

**Pull request merged**